elf2flt.c

```diff
@@ -1936,7 +1941,9 @@ int main(int argc, char *argv[])
 
   /* Read in all data sections.  */
   for (s = abs_bfd->sections; s != NULL; s = s->next)
-    if (s->flags & SEC_DATA) 
+    if ((s->flags & SEC_DATA) &&
+       ((s->flags & (SEC_READONLY | SEC_RELOC)) !=
```

Comment:
i think i'm following why this is correct, but since the code is a bit nuanced (which is an understatement for this entire code base), can you update the comment here to add a bit more detail ?  maybe something like `/* Read in all data sections that haven't already been collected.  */` ?

Reply:
Okay, comments in code expanded a bit in new force-pushed commit.